src/components/settingsModal/settingsModal.tsx

```diff
@@ -11,30 +12,55 @@ interface Permissions {
   [key: string]: boolean;
 }
 
+interface AudioDeviceSelection {
+  devices: string[];
+  selected: string;
+}
+
 const SettingsModal: React.FC<ModalProps> = ({ show, onClose }) => {
   if (!show) return null;
 
-  const [toggles, setToggles] = useState<Permissions>({})
+  const [permissions, setPermissions] = useState<Permissions>({})
+  const [audioInputDeviceSelection, setAudioInputDeviceSelection] = useState<AudioDeviceSelection>({devices: [], selected: ""})
+  const [audioOutputDeviceSelection, setAudioOutputDeviceSelection] = useState<AudioDeviceSelection>({devices: [], selected: ""})
+  const [inputDeviceSelected, setInputDeviceSelected] = useState<String>("")
+  const [outputDeviceSelected, setOutputDeviceSelected] = useState<String>("")
+
+  // get all info needed for the settings modal
   useEffect(() => {
     invoke("get_permissions").then((permissions: any) => {
-      setToggles(permissions as Permissions)
+      setPermissions(permissions as Permissions)
     })
+
+    // refresh audio input and ouput devices every 2 seconds
+    const refreshAudioDevices = setInterval(() => {
+      invoke("get_audio_input_devices").then((audioInputDeviceSelection: any) => {
+        setInputDeviceSelected(audioInputDeviceSelection.selected)
+        setAudioInputDeviceSelection(audioInputDeviceSelection)
+        // console.log(audioInputDeviceSelection.selected)
+      })
+  
+      invoke("get_audio_output_devices").then((audioOutputDeviceSelection: any) => {
+        setOutputDeviceSelected(audioOutputDeviceSelection.selected)
+        setAudioOutputDeviceSelection(audioOutputDeviceSelection)
+        // console.log(audioOutputDeviceSelection.selected)
+      })
+    }, 2000)
+
+    // this isn't working correctly, interval remains even after modal is not shown
+    return () => clearInterval(refreshAudioDevices)
```

Comment:
If I were to open and close the settings modal many times would we run into memory/performance issues?  I wonder if we can use something similar to what we have in App.tsx for activating the listeners for the backend

Reply:
yes it would, this needs fixed before merge. the intervals will all run independently and are never properly released. I think it has something to do with how we are using the show variable